src/library/scala/collection/Seq.scala

```diff
@@ -839,7 +839,15 @@ trait SeqOps[+A, +CC[_], +C] extends Any
     */
   @inline final def lengthIs: IterableOps.SizeCompareOps = new IterableOps.SizeCompareOps(this)
 
-  override def isEmpty: Boolean = lengthCompare(0) == 0
+  /** @inheritdoc
+   *
+   *  Note: Implementation uses `lengthCompare`. This method must be overridden in subclasses of
+   *  [[LinearSeqOps]], where `lengthCompare` uses `isEmpty`. To break that loop for the case of
+   *  an ill-constructed class, this implementation uses `knownSize` for instances of [[LinearSeq]].
+   */
+  override def isEmpty: Boolean =
+    if (this.isInstanceOf[LinearSeq[_]]) knownSize <= 0
```

Comment:
IMO it deosn't make sense to "pollute" this implementation with a special case for `LinearSeq` that's not even correct. An SOE is probably a worse failure mode, but still I don't think it's worth it.

Reply:
I don't disagree.